types/signed_header.go

```diff
@@ -93,6 +92,31 @@ var (
 	ErrSignatureVerificationFailed = errors.New("signature verification failed")
 )
 
+// VerifyCentralizedSequencer checks if a SignedHeader came from the expected centralized sequencer, given a genesis file.
+func (sh *SignedHeader) VerifyCentralizedSequencer(genesis *cmtypes.GenesisDoc) error {
```

Comment:
no need for this method, just check the public key bytes in the callsite itself (block manager). also, note that signed_header is self contained and we don't want methods that require additional context, e.g., genesis in this case.